src/lib/list/selection-list.ts

```diff
@@ -0,0 +1,187 @@
+import {merge} from 'rxjs/observable/merge';
+
+export class MdSelectionListBase {}
+export const _MdSelectionListMixinBase = mixinDisabled(MdSelectionListBase);
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-selection-list, mat-selection-list',
+  inputs: ['disabled'],
+  host: {
+    'role': 'listbox',
+    '[attr.tabindex]': '_tabIndex',
+    'class': 'mat-selection-list',
+    '(focus)': 'focus()',
+    '(keydown)': 'keydown($event)',
+    '[attr.aria-disabled]': 'disabled.toString()'},
+  template: '<ng-content></ng-content>',
+  styleUrls: ['list.css'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MdSelectionList extends _MdSelectionListMixinBase
+  implements Focusable, CanDisable, AfterContentInit, OnDestroy {
+  private _disableRipple: boolean = false;
+
+  /** Tab index for the selection-list. */
+  _tabIndex = 0;
+
+  /** Subscription to tabbing out from the selection-list. */
+  private _tabOutSubscription: Subscription;
+
+  /** Subscription to all list options' onFocus events */
+  private _optionsChangeSubscriptionOnFocus: Subscription;
+
+  /** Subscription to all list options' destroy events  */
+  private _optionsChangeSubscriptionDestroy: Subscription;
+
+  /** The FocusKeyManager which handles focus. */
+  _keyManager: FocusKeyManager;
+
+  /** The option components contained within this selection-list. */
+  @ContentChildren(MdListOption) options;
+
+  /** options which are selected. */
+  selectedOptions: SelectionModel<MdListOption> = new SelectionModel<MdListOption>(true);
+
+  /**
+   * Whether the ripple effect should be disabled on the list-items or not.
+   * This flag only has an effect for `mat-selection-list` components.
+   */
+  @Input()
+  get disableRipple() { return this._disableRipple; }
+  set disableRipple(value: boolean) { this._disableRipple = coerceBooleanProperty(value); }
+
+  constructor(private _element: ElementRef) {
+    super();
+  }
+
+  ngAfterContentInit(): void {
+    this._keyManager = new FocusKeyManager(this.options).withWrap();
+
+    if (this.disabled) {
+      this._tabIndex = -1;
+    }
+
+    this._optionsChangeSubscriptionOnFocus = this.onFocusSubscription();
+    this._optionsChangeSubscriptionDestroy = this.onDestroySubscription();
+  }
+
+  ngOnDestroy(): void {
+    if (this._tabOutSubscription) {
+      this._tabOutSubscription.unsubscribe();
+    }
+
+    if (this._optionsChangeSubscriptionDestroy) {
+      this._optionsChangeSubscriptionDestroy.unsubscribe();
+    }
+
+    if (this._optionsChangeSubscriptionOnFocus) {
+      this._optionsChangeSubscriptionOnFocus.unsubscribe();
+    }
+  }
+
+  focus() {
+    this._element.nativeElement.focus();
+  }
+
+  /**
+   * Map all the options' destroy event subscriptions and merge them into one stream.
+   */
+  onDestroySubscription(): Subscription {
+    let subscription = RxChain.from(this.options.changes)
```

Comment:
Probably don't need this variable; can return directly. Same thing for the focus subscription.

```ts
return RxChain.from(this.options.changes)...
```